apps/condo/domains/notification/adapters/firebaseAdapter.js

```diff
@@ -226,6 +228,23 @@ class FirebaseAdapter {
                                 pushType: get(pushTypes, notifications[idx].token, null),
                             })
                     )
+                    // handling expired token error. https://firebase.google.com/docs/cloud-messaging/manage-tokens?hl=ru#detect-invalid-token-responses-from-the-fcm-backend
+                    for (const res of fbResult.responses) {
+                        const context = getSchemaCtx('RemoteClient')
+                        if (get(res, 'error.code') === 'messaging/registration-token-not-registered') {
+                            const field = isVoIP ? 'pushTokenVoIP' : 'pushToken'
+                            const [remoteClient] = await find('RemoteClient', {
```

Comment:
You should not use anything related to data inside the adapter.

Reply:
You need to design the adapter in a way that it can be used in any other project. Its sole purpose is to handle delivery tasks. It should not make any requests, as that would violate the principle of abstraction.